packages/zql/src/ivm/view-apply-change.ts

```diff
@@ -7,14 +7,51 @@ import {
 } from '../../../shared/src/asserts.ts';
 import {must} from '../../../shared/src/must.ts';
 import type {Row} from '../../../zero-protocol/src/data.ts';
-import type {Change} from './change.ts';
-import {drainStreams, type Comparator} from './data.ts';
+import {drainStreams, type Comparator, type Node} from './data.ts';
 import type {SourceSchema} from './schema.ts';
 import type {Entry, EntryList, Format} from './view.ts';
 
+/**
+ * `applyChange` does not consume the `relationships` of `ChildChange#node`,
+ * `EditChange#node` and `EditChange#oldNode`.  The `ViewChange` type
+ * documents and enforces this via the type system.
+ */
+export type ViewChange =
```

Comment:
I don't love having these new types exposed in `@rocicorp/zero/advanced`, but it seemed better than just dropping the relationships from ChildChange and EditChange.